commands/issue.go

```diff
@@ -48,80 +56,146 @@ var (
+				maxNumWidth = numWidth
 			}
 		}
-	})
-}
 
-func createIssue(cmd *Command, args *Args) {
-	runInLocalRepo(func(localRepo *github.GitHubRepo, project *github.Project, gh *github.Client) {
-		if args.Noop {
-			ui.Printf("Would create an issue for %s\n", project)
-		} else {
-			title, body, err := getTitleAndBodyFromFlags(flagIssueMessage, flagIssueFile)
-			utils.Check(err)
-
-			if title == "" {
-				title, body, err = writeIssueTitleAndBody(project)
-				utils.Check(err)
+		colorize := ui.IsTerminal(os.Stdout)
+
+		for _, issue := range issues {
+			if issue.PullRequest != nil {
+				continue
+			}
+
+			num := fmt.Sprintf("#%d", issue.Number)
+			numWidth := len(num)
+
+			if colorize {
+				issueColor := 32
+				if issue.State == "closed" {
+					issueColor = 31
+				}
+				num = fmt.Sprintf("\033[%dm%s\033[0m", issueColor, num)
 			}
 
-			issue, err := gh.CreateIssue(project, title, body, flagIssueLabels)
-			utils.Check(err)
+			ui.Printf("%*s%s  %s", maxNumWidth+1-numWidth, "", num, issue.Title)
+
+			for i, label := range issue.Labels {
+				color, err := utils.NewColor(label.Color)
+				if err != nil {
+					utils.Check(err)
+				}
+
+				textColor := 16
+				if color.Brightness() < 0.65 {
+					textColor = 15
+				}
+
+				if i == 0 {
+					ui.Printf(" ")
+				}
+				ui.Printf(" \033[38;5;%d;48;2;%d;%d;%dm %s \033[m", textColor, color.Red, color.Green, color.Blue, label.Name)
+			}
 
-			ui.Println(issue.HTMLURL)
+			ui.Printf("\n")
 		}
-	})
+	}
+
+	os.Exit(0)
 }
 
-func writeIssueTitleAndBody(project *github.Project) (string, string, error) {
-	message := `
-# Creating issue for %s.
+func createIssue(cmd *Command, args *Args) {
+	localRepo, err := github.LocalRepo()
+	utils.Check(err)
+
+	project, err := localRepo.MainProject()
+	utils.Check(err)
+
+	gh := github.NewClient(project.Host)
+
+	var title string
+	var body string
+	var editor *github.Editor
+
+	if cmd.FlagPassed("message") {
+		title, body = readMsg(flagIssueMessage)
+	} else if cmd.FlagPassed("file") {
+		title, body, err = readMsgFromFile(flagIssueFile)
+		utils.Check(err)
+	} else {
+		cs := git.CommentChar()
+		message := strings.Replace(fmt.Sprintf(`
+# Creating an issue for %s
 #
 # Write a message for this issue. The first block of
 # text is the title and the rest is the description.
-`
-	message = fmt.Sprintf(message, project.Name)
+`, project), "#", cs, -1)
+
+		editor, err := github.NewEditor("ISSUE", "issue", message)
+		utils.Check(err)
+
+		title, body, err = editor.EditTitleAndBody()
+		utils.Check(err)
+	}
 
-	editor, err := github.NewEditor("ISSUE", "issue", message)
-	if err != nil {
-		return "", "", err
+	if title == "" {
+		utils.Check(fmt.Errorf("Aborting creation due to empty issue title"))
 	}
 
-	defer editor.DeleteFile()
+	params := &github.IssueParams{
+		Title:  title,
+		Body:   body,
+		Labels: flagIssueLabels,
+	}
+
+	if args.Noop {
+		ui.Printf("Would create issue `%s' for %s\n", params.Title, project)
```

Comment:
I don't feel that the `--noop` output should be absolutely detailed here. The `--noop` flag is mostly useful for other hub commands that result in a git command being executed because it allows you to inspect the final git command.